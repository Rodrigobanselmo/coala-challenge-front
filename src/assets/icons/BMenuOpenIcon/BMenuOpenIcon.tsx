import React, { FC } from "react";

import { IIconMuiProps } from "@/core/interfaces/IIconMuiProps";
import MenuOpenIcon from "@mui/icons-material/MenuOpen";

export const BMenuOpenIcon: FC<IIconMuiProps> = ({ ...props }) => {
  return <MenuOpenIcon {...props} />;
};

export default MenuOpenIcon;
