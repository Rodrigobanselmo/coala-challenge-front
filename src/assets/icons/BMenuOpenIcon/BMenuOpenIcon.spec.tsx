import React from "react";
import { render, screen } from "@testing-library/react";
import { BMenuOpenIcon } from "./BMenuOpenIcon";

describe("BMenuOpenIcon", () => {
  it("renders without errors", () => {
    render(<BMenuOpenIcon />);
    const icon = screen.getByTestId("MenuOpenIcon");

    expect(icon).toBeInTheDocument();
    expect(icon).toMatchSnapshot();
  });

  it("applies custom props", () => {
    render(<BMenuOpenIcon fontSize="large" color="primary" />);

    const icon = screen.getByTestId("MenuOpenIcon");
    expect(icon).toMatchSnapshot();
  });
});
