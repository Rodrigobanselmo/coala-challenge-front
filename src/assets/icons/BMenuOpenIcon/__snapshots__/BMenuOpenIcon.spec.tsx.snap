// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BMenuOpenIcon applies custom props 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge css-bsqp9w-MuiSvgIcon-root"
  data-testid="MenuOpenIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M3 18h13v-2H3v2zm0-5h10v-2H3v2zm0-7v2h13V6H3zm18 9.59L17.42 12 21 8.41 19.59 7l-5 5 5 5L21 15.59z"
  />
</svg>
`;

exports[`BMenuOpenIcon renders without errors 1`] = `
<svg
  aria-hidden="true"
  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
  data-testid="MenuOpenIcon"
  focusable="false"
  viewBox="0 0 24 24"
>
  <path
    d="M3 18h13v-2H3v2zm0-5h10v-2H3v2zm0-7v2h13V6H3zm18 9.59L17.42 12 21 8.41 19.59 7l-5 5 5 5L21 15.59z"
  />
</svg>
`;
